zuul_alpha/core.py

```diff
@@ -238,28 +227,15 @@ def _encrypt_all_secrets(self, secrets_json):
             log.info(secret_name)
             self._encrypt(secret_name, secret)
 
-    def _decrypt_file(self, secret_file):
-        '''Decrypts a RSA encrypted file.'''
-        try:
-            with open(secret_file) as f:
-                    secret = self.decryptor.decrypt(b64decode(f.read()))
-        except Exception as e:
-            log.error(
-                "Could not decrypt secret '%s', returning '' (Reason: %s)" % (
-                    os.path.basename(secret_file), e.message))
-            secret = None
-
-        return secret
-
     def _decrypt_aes_wrapped_file(self, secret_file):
-        '''Decrypts a binary which contains an RSA encrypted AES session key
-        and AES encrypted data.'''
+        '''Decrypts a binary file which contains an RSA encrypted AES session
+        key and AES encrypted data.'''
         try:
             with open(secret_file, 'r') as f:
                 ciphertext_bin = b64decode(f.read())
 
             # break out bin file
-            offset = self.rsa_size_in_bytes
+            offset = (self.rsa_key_size / 8)
             encrypted_session_key = ciphertext_bin[:offset]
```

Comment:
Throw a check in here to verify offset is <= the file size. This isn't a bug atm because of python's handling of `[:]` but if this code is translated at some point to another language it would be easy to introduce a bug.